### ISSUE TITLE
Can't sync two distinct profiles at the same time

### ISSUE THREAD

**Issue by olivierlemoal** [borgbase/vorta#1037]
**Describe the bug**
I have two profiles with different sets of source folders / destination repository.
Both are scheduled to run every 3 hours. In the end, only one is backup, the other fails (see logs).

**Environment**
- OS: Arch
- Vorta version: 0.7.7
- Installed from: AUR pkg

**Additional context**
```
2021-07-13 21:00:00,000 - apscheduler.executors.default - INFO - Running job "VortaScheduler.create_backup (trigger: cron[hour='*/3', minute='0'], next run at: 2021-07-14 00:00:00 CEST)" (scheduled at 2021-07-13 21:00:00+02:00)
2021-07-13 21:00:00,004 - apscheduler.executors.default - INFO - Running job "VortaScheduler.create_backup (trigger: cron[hour='*/3', minute='0'], next run at: 2021-07-14 00:00:00 CEST)" (scheduled at 2021-07-13 21:00:00+02:00)
2021-07-13 21:00:00,008 - vorta.scheduler - INFO - Starting background backup for photos
2021-07-13 21:00:00,010 - vorta.notifications - DEBUG - notification not suppressed
2021-07-13 21:00:00,011 - vorta.scheduler - INFO - Starting background backup for config_k3s_lan
2021-07-13 21:00:00,017 - vorta.notifications - DEBUG - notification not suppressed
2021-07-13 21:00:00,020 - vorta.keyring.abc - DEBUG - Only available on macOS
2021-07-13 21:00:00,021 - vorta.borg.borg_thread - DEBUG - Using VortaDBKeyring keyring to store passwords.
2021-07-13 21:00:00,024 - vorta.scheduler - INFO - Preparation for backup successful.
2021-07-13 21:00:00,037 - vorta.borg.borg_thread - INFO - Running command /usr/bin/borg create --list --progress --info --log-json --json --filter=AM -C lz4 xxxxxxx.repo.borgbase.com:repo::my-hostname-2021-07-13-210000 /path/to/backup1
2021-07-13 21:00:00,052 - vorta.scheduler - ERROR - Conditions for backup not met. Aborting.
2021-07-13 21:00:00,052 - vorta.scheduler - ERROR - Backup is already in progress.
2021-07-13 21:00:00,054 - vorta.notifications - DEBUG - notification not suppressed
2021-07-13 21:00:00,068 - apscheduler.executors.default - INFO - Job "VortaScheduler.create_backup (trigger: cron[hour='*/3', minute='0'], next run at: 2021-07-14 00:00:00 CEST)" executed successfully
2021-07-13 21:00:03,801 - vorta.borg.borg_thread - INFO - Creating archive at "xxxxxxx.repo.borgbase.com:repo::my-hostname-2021-07-13-210000"
2021-07-13 21:00:04,062 - vorta.borg.borg_thread - INFO - Remote: Storage quota: 13.49 GB out of 399.92 GB used.
2021-07-13 21:00:04,759 - vorta.borg.borg_thread - INFO - Remote: Storage quota: 13.49 GB out of 399.92 GB used.
2021-07-13 21:00:05,129 - vorta.notifications - DEBUG - notification not suppressed
2021-07-13 21:00:05,136 - vorta.scheduler - INFO - Backup creation successful.
2021-07-13 21:00:05,144 - vorta.scheduler - INFO - Doing post-backup jobs for photos
2021-07-13 21:00:05,149 - vorta.scheduler - INFO - Finished background task for profile photos
```

**Comment by olivierlemoal** [comment]
Looking at the `BorgThread` class, there is mutex preventing two backups running at the same time.

**Comment by bastiencyr** [comment]
On my laptop, if I run two backups at the same time, the two backups are not executed and its not possible after to execute any scheduling task with the following error : "skipped: maximum number of running instances reached (1)".

> Edit: just found about this discussion #830

As it is suggested in this discussion, a queue may be a good idea. Actually, multiple queues (one queue by repo) can be implemented.

**Comment by m3nu** [comment]
Yeah, a queue could be useful if multiple tasks are requested. At least if it's a scheduled background task. For user-initiated tasks, I'm ok with failing them.

**Comment by bastiencyr** [comment]
Maybe we can close this issue now @m3nu ?

**Comment by m3nu** [comment]
Sure. You should have permission to close it too.

Fixed by #1045
